src/mon-user/mon-user.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { CreateMonUserDto } from './dto/create-mon-user.dto';
+import { MonUser } from './schemas/mon-user.schema';
+
+@Injectable()
+export class MonUserService {
+  constructor(
+    @InjectModel(MonUser.name) private readonly monUserModel: Model<MonUser>,
+  ) {}
+
+  async create(createCatDto: CreateMonUserDto): Promise<MonUser> {
+    const createdCat = await this.monUserModel.create(createCatDto);
+    return createdCat;
+  }
+
+  async findAll(): Promise<MonUser[]> {
```

Comment:
Better to use interfaces in services and DTO only in controller
Using this approach we don't overheat nest application runtime because interfaces are weightless in a runtime, but classes has